### ISSUE TITLE
12 word phrases

### ISSUE THREAD

**Issue by kaze504** [gurnec/btcrecover#416]
Anyone know how to recover 12 word phrases of enjin wallet?? Please help me ..

**Comment by 3rdIteration** [comment]
https://btcrecover.readthedocs.io/en/latest/Usage_Examples/basic_seed_recoveries/

**Comment by kirokou** [comment]
Hello how use this command if i don't have any adrss ?
I juste have the first 22 words and i lost the 2 lasts word. 
It's ledger and i work with mac os
Please help me...

python seedrecover.py --wallet-type bip39 --addrs bc1qv87qf7prhjf2ld8vgm7l0mj59jggm6ae5jdkx2 --mnemonic "element entire sniff tired miracle solve shadow scatter hello never tank side sight isolate sister uniform advice pen praise soap lizard festival connect" --addr-limit 5

**Comment by nicoschtein** [comment]
Hi kirokou!

You’ve shared your seed on a public website. I’ve recovered your seed,
please share your email and I’ll send it over.
I just hope you have a passphrase with this seed !

**Comment by 3rdIteration** [comment]
@kirokou 

I just had a look and that's a test seed I used back in 2019, so at least they didn't share the whole thing publicly. :)
 
The last two words missing is no problem and not having an address is fine if you ever used the device with Bitcoin, Eth, LTC, etc... You can just use an address database. You will need to use my updated fork for this https://github.com/3rdIteration/btcrecover as this functionality is broken in Gurnec's repo.

I have a video here: https://youtu.be/6vw2x053LW4 and documentation here: https://btcrecover.readthedocs.io/en/latest/Creating_and_Using_AddressDB/

**Comment by nicoschtein** [comment]
Haha I now think it’s the example seed form the docs ! :facepalm:

Kirokou must have been asking for guidance and used that form the docs

**Comment by kirokou** [comment]
Hello!
 yes indeed it is the example of the documentation. I wouldn't dare post the real passphrase online.
Having no public address, I have:
(1) download the BTC address database on mega. https://mega.nz/folder/k4E1BahC#iPYiM7D7ZVsjhcFdohDosA)
(2) unpack the addresses-BTC.db in the root folder
(3) I then placed the following command: 
`python3 seedrecover.py --no-dupchecks --typos 2 --big-typos 2 --addr-limit 10 --bip32-path "m / 44 '/ 2' / 0 '/ 0 "--addressdb addresses-BTC.db --wallet-type bip39`
(4) then I enter the passphrase you used in the demo video: "element entire sniff tired miracle solve shadow scatter hello never tank side sight isolate sister uniform advice pen praise soap lizard festival 23 24"
(5) But it doesn't work. I get the following error:
```
`Loading address database ...
Loaded 771514103 addresses from database ...
Using the 'en' wordlist.
'23' was in your seed, but there is no similar seed word.
'24' was in your seed, but there is no similar seed word.
2021-03-12 01:13:07 : Phase 1/1: up to 2 mistakes, 2 of which can be an entirely different seed word.
Counting passwords ...
Done                                                                          
2021-03-12 01:13:21 : Using 4 worker threads
4194304 of 4194304 [#######################################################################################################################] 0:02:15, 
2021-03-12 01:15:38 : Search Complete
 Seed not found, sorry...
```

**Comment by kirokou** [comment]
hello. Help please.

**Comment by 3rdIteration** [comment]
Firstly, there was an issue with the AddressDB on there so I have reuploaded another copy as of yesterday.

Secondly, you are using a BTC addressDb with Litecoin derivation path, so that won't work. I would suggest that you drop all the additional parameters other than the one required for the addressDB, as the default should handle 2 missing words just fine.

**Comment by kirokou** [comment]
It is not possible to unzip the new btc database file that you added on March 13th. After downloading, unable to unzip. I received the same error on ios and on linux. Did you try?

**Comment by 3rdIteration** [comment]
Yep, unzips fine on in both 7-zip and the unzip tool built in to Windows. I'll try re-downloading. What's the sha256sum of the zipped file? (It should be: AE1683EA85447A22E6BB19B0A355EB5E2C2B307B01B19E56C43397FB06169A23)

**Comment by 3rdIteration** [comment]
I just re-downloaded it from Mega and it unzips fine, has the correct checksum.

**Comment by kirokou** [comment]
@3rdIteration Thank you. 
with new BTC database, i have 
python3 seedrecover.py --no-dupchecks --addressdb addresses-BTC.db

Loading address database ...
Loaded 771514103 addresses from database ...
objc[56470]: Class FIFinderSyncExtensionHost is implemented in both /System/Library/PrivateFrameworks/FinderKit.framework/Versions/A/FinderKit (0x7fffad1bd3d8) and /System/Library/PrivateFrameworks/FileProvider.framework/OverrideBundles/FinderSyncCollaborationFileProviderOverride.bundle/Contents/MacOS/FinderSyncCollaborationFileProviderOverride (0x10c150f50). One of the two will be used. Which one is undefined.
Using the 'en' wordlist.
Seed sentence was too long, deleting 1 word from each guess.
2021-03-14 19:10:43 : Phase 1/4: up to 2 mistakes, excluding entirely different seed words.
2021-03-14 19:10:44 : Using 4 worker threads
4642 of 4642 [##############################################################################################################################] 0:00:02, 
2021-03-14 19:10:47 : Search Complete
 Seed not found
2021-03-14 19:10:47 : Phase 2/4: 1 mistake which can be an entirely different seed word.
2021-03-14 19:10:48 : Using 4 worker threads
22 of 22 [##################################################################################################################################] 0:00:02, 
2021-03-14 19:10:50 : Search Complete
 Seed not found
2021-03-14 19:10:50 : Phase 3/4: up to 2 mistakes, 1 of which can be an entirely different seed word.
2021-03-14 19:10:56 : Using 4 worker threads
946176 of 946176 [##########################################################################################################################] 0:00:34, 
2021-03-14 19:11:31 : Search Complete
 Seed not found
2021-03-14 19:11:31 : Phase 4/4: up to 2 mistakes, 2 of which can be an entirely different seed word.
Counting passwords ...
Done                                                                          
2021-03-14 19:40:21 : Using 4 worker threads
946176 of 946176 [##########################################################################################################################] 0:29:45, 
2021-03-14 20:10:07 : Search Complete
 Seed not found, sorry...

**Comment by kirokou** [comment]
I do not understand.
 Installation of btcrecover is ok. I upload the new BTC address, I have the 22 words,
 I launched the basic command: python3 seedrecover.py --no-dupchecks --addressdb addresses-BTC.db
But seed not found. I lose hope.

**Comment by 3rdIteration** [comment]
It says that the sentence is too long, so the issue is that it is trying to automatically guess the length of the mnemonic and selecting 21. If you specify the mnemonic length of 24 it should work.

It's a straightforward recovery, if you are totally stuck email me at steve@cryptoguide.tips and we can to a paid consult or assisted recovery.

**Comment by btcexchanging** [comment]
if you haven't recovered your wallet, i can help you recover your wallet.
I have recovered most of the common wallets and recovery phrases for many users
you can reach me here at my telegram account:  @walletrecover

**Comment by anywalletrecovery** [comment]
hi
i might be able to help you recover your password
send me an email here: anywalletrecovery@gmail.com

**Comment by viladmir2022** [comment]
hi, if you still have wallet problem. I can recover your lost wallet.
I have powerful servers and excellent software that can find any password.
and my fee is only 15-25% of the recovered fund
here is my telegram id: @viladmir2022  and Gmail: walletsrecoveryservice@gmail.com

**Comment by johnjames333** [comment]
we have developed our own hardware for lost words superfast speed
and if you still has problem I can help you run it. you can find my contact details on my profile
or you can send me message in telegram: @johnjames333
